app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/InMemoryRepository.kt

```diff
@@ -0,0 +1,58 @@
+package com.github.HumanLearning2021.HumanLearningApp.model
+
+import android.util.Base64
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlin.random.Random
+
+/** In-memory, impermanent key-value storage.
+ *
+ */
+class InMemoryRepository<V>(
+    private val random: Random = Random,
+) : Repository<String, V> {
+
+    private inner class Context {
+        var data = mutableMapOf<String, V>()
+    }
+
+    private val mutex = Mutex()
+    private val ctx = Context()
+    private suspend fun <R> withContext(action: Context.() -> R): R = mutex.withLock {
+        ctx.action()
+    }
+
+    suspend fun createWith(fkv: (String) -> V) = withContext {
+        var k: String
+        do {
+            k = Base64.encodeToString(random.nextBytes(15), Base64.URL_SAFE)!!
```

Comment:
```suggestion
            k = Base64.encodeToString(random.nextBytes(15), Base64.URL_SAFE)
```

Reply:
The compiler complains about nullability inference. We have to explicitly assert that `encodeToString` does not return null.

Reply:
I don't understand because according to the [doc](https://developer.android.com/reference/android/util/Base64#encodeToString(byte[],%20int) the return value is already non-null and compiling on my machine did not throw any warnings ?

Reply:
> I don't understand because according to the [doc](https://developer.android.com/reference/android/util/Base64#encodeToString(byte%5B%5D,%20int)) the return value is already non-null and compiling on my machine did not throw any warnings ?

It does for me too in the latest commit, my bad.